### PR TITLE
KOGITO-5326 Setup deploy/promote notifications on demand

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5326

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/212
- https://github.com/kiegroup/kogito-runtimes/pull/1449
- https://github.com/kiegroup/optaplanner/pull/1436
- https://github.com/kiegroup/kogito-apps/pull/945
- https://github.com/kiegroup/kogito-examples/pull/800
- https://github.com/kiegroup/kogito-images/pull/602
- https://github.com/kiegroup/kogito-operator/pull/944
- https://github.com/kiegroup/rhpam-kogito-operator/pull/56

### CODE REVIEW

**Comment by radtriste** [comment]
Jenkins retest this

**Review by Ginxo — APPROVED** [review]
LGTM

File: Jenkinsfile.release [hunk 1]
```diff
@@ -32,9 +32,10 @@ baseImageParamsPrefix = 'BASE_IMAGE'
 promoteImageParamsPrefix = 'PROMOTE_IMAGE'
 
 pipeline {
-    agent {
-        label 'kie-rhel7 && !master'
-    }
+    // agent {
+    //     label 'kie-rhel7 && !master'
+    // }
+    agent any
```
Comment (Ginxo) [inline Jenkinsfile.release]: is `any` expected here? I don't think so :thinking:

File: Jenkinsfile.nightly [hunk 2]
```diff
@@ -334,15 +332,15 @@ pipeline {
 def buildJob(String jobName, List buildParams, boolean propagate=true) {
     echo "Build ${jobName} with params ${buildParams} and propagate = ${propagate}"
 
-    def job = build(job: "${jobName}", wait: true, parameters: buildParams, propagate: propagate)
-    JOBS[jobName] = job
+    // def job = build(job: "${jobName}", wait: true, parameters: buildParams, propagate: propagate)
```
Comment (Ginxo) [inline Jenkinsfile.nightly]: to be removed?

File: Jenkinsfile.nightly [hunk 3]
```diff
@@ -334,15 +332,15 @@ pipeline {
...
+    // JOBS[jobName] = job
 
-    // Set Unstable if we don't propagate and job failed
-    if (!propagate && !isJobSucceeded(jobName)) {
-        addUnstableStage(jobName)
-        unstable("Job ${jobName} finished with result ${job.getResult()}")
-    }
-    return job
+    // // Set Unstable if we don't propagate and job failed
```
Comment (Ginxo) [inline Jenkinsfile.nightly]: to be removed? (all commented lines I mean)

File: Jenkinsfile.release [hunk 4]
```diff
@@ -456,36 +457,36 @@ pipeline {
...
-        if (result['ACTION'] == 'retry') {
-            return buildJob(jobName, buildParams)
-        } else if (result['ACTION'] == 'continue') {
-            registerJob(jobName, job)
-        }
-    } else {
-        // Succeeded
-        registerJob(jobName, job)
-    }
+    // def job = build(job: "./${jobName}", wait: true, parameters: buildParams, propagate: false)
```
Comment (Ginxo) [inline Jenkinsfile.release]: to be removed?

**Review by Ginxo — CHANGES_REQUESTED** [review]
please review

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**